### ISSUE TITLE
Incompatibility with pthreads

### ISSUE THREAD

**Issue by LuckyFellow** [php-decimal/ext-decimal#12]
Environment
==========

  * PHP: 7.2.14 (cli) ( ZTS )
  * decimal: 1.1.1
  * pthreads:  3.2.0 (http://php.net/manual/en/book.pthreads.php / https://github.com/krakjoe/pthreads)
  * libmpdec: 2.4.2
  * OS: docker, debian:stretch

Summary
======

Without threads, operations work as expected.
As soon as operations are performed in a seperate thread, some result in "INF".

Reproducing Code
==============

```
<?php

$thread = new class() extends Thread {

    public function run() {
        var_dump((new Decimal\Decimal("1"))->mul("9"));
        var_dump((new Decimal\Decimal("1"))->mul("10"));
    }
};

$thread->start() && $thread->join();
?>

```
Expected Output
=============

```
object(Decimal\Decimal)#39 (2) {
  ["value"]=>
  string(1) "9"
  ["precision"]=>
  int(28)
}
object(Decimal\Decimal)#42 (2) {
  ["value"]=>
  string(2) "10"
  ["precision"]=>
  int(28)
}

```
Actual Output
============

```
object(Decimal\Decimal)#3 (2) {
  ["value"]=>
  string(1) "9"
  ["precision"]=>
  int(28)
}
PHP Warning:  Loss of data on string conversion in /app/test.php on line 7
object(Decimal\Decimal)#2 (2) {
  ["value"]=>
  string(3) "INF"
  ["precision"]=>
  int(28)
}

```

**Comment by rtheunissen** [comment]
Fixed in 48920be1b066d04308c7a271808bc081a632e62b, will confirm that it is safe and correct, then release a patch.

**Comment by rtheunissen** [comment]
Released as v1.1.2 :sparkles:

**Comment by rtheunissen** [comment]
I'll get on that asap, thanks for the report.

**Comment by krakjoe** [comment]
Hi @rtheunissen ... I read the libmpdec source, it actually does require that contexts are truly global, I committed to both your active branches the fixes for that, hope I didn't make a mess of logs or do it badly ...

@LuckyFellow leaving this for you to close ...

**Comment by rtheunissen** [comment]
@krakjoe  @LuckyFellow I've fixed this a little bit differently in b2114856d04ee88ad375538a116896258ac288b7. Will wait for confirmation before we close this. :+1:

**Comment by LuckyFellow** [comment]
I can confirm, that there is no warning issued anymore and the operations function properly.

Thank you very much!
